app/src/org/commcare/android/tasks/DataPullTask.java

```diff
@@ -133,7 +133,7 @@ public DataPullTask(String username, String password, String server, String keyP
     protected void onCancelled() {
         super.onCancelled();
         if(wasKeyLoggedIn) {
-            CommCareApplication._().logout();
+            CommCareApplication._().getSession().startLogout();

```

Comment:
Leaving a note for myself to come back to this.

The datapull task "login" ops are more about seating a user DB than "logging in", but this might be equivalent. Will evaluate.